src/clj_pgp/core.clj

```diff
@@ -258,9 +259,16 @@
   "Lazily decodes a sequence of PGP objects from an input stream."
   [^InputStream input]
   (let [factory (PGPObjectFactory. input (BcKeyFingerprintCalculator.))]
-    (->>
-      (repeatedly #(.nextObject factory))
-      (take-while some?))))
+    (->> (range)
+         (map (fn [n]
```

Comment:
Now that this is no longer anonymous, please give this a name so that exceptions which do get thrown will have clearer stack traces.